nixos/modules/system/boot/luksroot.nix

```diff
@@ -254,8 +254,27 @@ let
         for try in $(seq 3); do
             ${optionalString yubikey.twoFactor ''
             echo -n "Enter two-factor passphrase: "
-            read -r k_user
-            echo
+            k_user=
+            while true; do
+                if [ -e /crypt-ramfs/k_user ]; then
+                    echo "reused"
+                    k_user=$(cat /crypt-ramfs/k_user)
+                    break
+                else
+                    # Try reading it from /dev/console with a timeout
+                    IFS= read -t 1 -r k_user
+                    if [ -n "$k_user" ]; then
+                       ${if luks.reusePassphrases then ''
+                         # Remember it for the next device
+                         echo -n "$k_user" > /crypt-ramfs/k_user
+                       '' else ''
+                         # Don't save it to ramfs. We are very paranoid
```

Comment:
(unrelated)

paranoid about someone dousing your computer in liquid nitrogen and taking out all the RAM? if an attacker has this much liquid nitrogen you got much bigger problems :laughing:

Reply:
The comment about about being paranoid (as well as the corresponding code) was borrowed from similar function `do_open_gpg_card`—which in turn was borrowed from `do_open_passphrase`. Paranoia must run deep within our community :laughing: